src/types/opcode.ts

```diff
@@ -0,0 +1,43 @@
+export type Variable = {
+  name: string;
+  description: string;
+  expression?: string;
+};
+
+type Memory = {
+  before?: string;
+  after: string;
+};
+
+export type Example = {
+  input?: string | string[];
+  output?: string;
+  memory?: Memory;
+};
+
+type GasComputation = {
+  name: string;
+  description: string;
+  expression: string;
+  variables: Variable[];
+};
+
+export type Reference = {
+  name: string;
+  url: string;
+};
+
+export type Opcode = {
+  number: number;
+  name: string;
+  description: string;
+  minGas: number;
+  gasComputation?: GasComputation;
+  inputs?: Variable[];
+  outputs?: Variable[];
+  examples: Example[];
+  playgroundLink?: string;
+  errorCases: string[];
+  notes?: string[];
+  references: Reference[];
```

Comment:
I was thinking of introducing `type Markdown = string`, and adding a simple markdown parser to the UI components. We can leave this as-is for this PR, but just wanted to share how I was thinking about things. Being able to use markdown in these arbitrary string fields (reference, notes, error cases, etc.) will give a lot more flexibility on how the UI side, while keeping a decent separation of concerns between the content itself and how it's rendered (it of course introduces some coupling but it seems like an ok tradeoff, still thinking about it though)

Reply:
Hey, that's something I was also thinking about! 😄 I'm not familiar with Markdown parser and stuff but this sounds like a great solution to have more flexibility.